codegen-service/llm-generation-core/src/main/java/org/rj/modelgen/llm/prompt/TemplatedPromptGenerator.java

```diff
@@ -0,0 +1,54 @@
+package org.rj.modelgen.llm.prompt;
+
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+
+public class TemplatedPromptGenerator<TImpl extends TemplatedPromptGenerator<?, TSelector>, TSelector> extends PromptGenerator<TImpl, TSelector> {
+    private final Map<TSelector, String> prompts;
+
+    public TemplatedPromptGenerator() {
+        this(Map.of());
```

Comment:
FYI: `Map.of()` actually returns an immutable impl of the Map interface, so you may hit errors later if you e.g. try to add new prompts.  I'd switch it for something like `new HashMap<>()`

Reply:
Addressed